server/src/assets/game/tableGenerator/converter.py

```diff
@@ -1,127 +1,184 @@
+            and int(route["id"]) not in blacklistedRouteIDs):
+
+            routeID = int(route["id"])
+            rLen = int(route["length"][:-5]) # length calc here ignores millimeters
+            # If this route is the first to this destination or shorter than the current one...
+            if (route["destination"] not in destsToShortestRouteWithLength.keys() 
+                or destsToShortestRouteWithLength[route["destination"]][2] > rLen):
+
+                destsToShortestRouteWithLength[route["destination"]] = (routeID, route["source"], rLen)
+                blockDictRet[route["destination"]] = {
+                    "route-id"    : routeID,
+                    "orientation" : route["orientation"],
+                    "block"       : route["sections"][-1]["id"],
+                    "segment"     : route["path"][-2]["id"]
+                }
+                # Remove "a" and "b" suffixes from "segment" entry if they exist
+                if route["path"][-2]["id"].endswith("a") or route["path"][-2]["id"].endswith("b"):
+                    blockDictRet[route["destination"]]["segment"] = route["path"][-2]["id"][:-1]
+    return blockDictRet
+
+
+def processConfiguration(config: str):
+    """
+    For a given configuration, loads config files, generates the destinations json and persists it as a file.
+    """
+    # Get the blacklisted route IDs(if any)
+    blacklistedRouteIDs = getBlacklistedRouteIDsForConfig(config)
+    # Get the interlockingTable and extrasConfig
+    try:
+        interlockingTable, extrasConfig = getInterlockingAndExtrasConfigFileContent(config)
+    except Exception as e:
+        print(e)
+        return False
+    # Get the list of signals with valid mappings (signalID appended with 'a' if its a composite signal)
+    signalFlagMapCSVFilepath = "{}/{}.csv".format(signalFlagCSVFileDirectory, config)
+    signalsWithDefMapping = getSignalIDStrsWithDefinedMapping(signalFlagMapCSVFilepath, extrasConfig)
+
+    print("Loaded config files for config " + config + ", now calculating routes")
+
+    collectedRoutesPerBlock = {}
+    # Relevant keys for searching in extrasConfig
+    blocktypes = ["blocks", "platforms"]
+    # For blocks of all kinds, get the routes to possible destinations.
+    for blockType in blocktypes:
+        for block in extrasConfig[blockType]:
+            # Initialize block dict, then calculate routes
+            collectedRoutesPerBlock[block["id"]] = {}
+            blockDict = calcWantedRoutesForBlock(block, interlockingTable, 
+                                                 signalsWithDefMapping, blacklistedRouteIDs)
+            # Now insert the routes into the "main" dict in the order that they appear in the 
+            # symbol/flag mappings csv file. The order in the CSV file is the same as the order 
+            # in signalsWithDefMapping, so we use that as the reference.
+            for sig in signalsWithDefMapping:
```

Comment:
Aren't the signals in `blockDict.keys()` a subset of `signalsWithDefMapping`? Would be more efficient to loop over `blockDict` instead.
```python
for sig in blockDict:
    if sig in signalsWithDefMapping:
        collectedRoutesPerBlock[block["id"]][sig] = blockDict[sig]
```

Reply:
I see, I should have read the comment more carefully